src/pytest_md/plugin.py

```diff
@@ -149,13 +150,13 @@ def create_summary(self):
 
         return summary + "\n\n" + outcome_text
 
-    def create_results(self):
+    def create_results(self) -> str:
         """Create results for the individual tests for the Markdown report."""
 
         outcomes = {}
 
         for outcome in (o for o in Outcome if o in self.reports):
-            reports_by_file = collections.defaultdict(list)
+            reports_by_file: Dict[Outcome, List] = collections.defaultdict(list)
```

Comment:
We group lists of reports by test file here:

```suggestion
            reports_by_file: Dict[str, List] = collections.defaultdict(list)
```